### PR TITLE
Dns discovery

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]

File: waku/src/node/discovery.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use url::Url;
+// internal
+use crate::general::JsonResponse;
+use crate::Result;
+
+/// RetrieveNodes returns a list of multiaddress given a url to a DNS discoverable ENR tree.
+/// The nameserver can optionally be specified to resolve the enrtree url. Otherwise uses the default system dns.
+pub fn waku_dns_discovery(
+    url: &Url,
+    server: Option<&str>,
```
Comment (Zeegomo) [inline waku/src/node/discovery.rs:15]: should this be typed instead of a plain str?

Comment (danielSanchezQ) [reply waku/src/node/discovery.rs:15]: I tried to find a proper type. But couldn't find/do not know about a proper type for it. Unless creating a type alias. Any suggestion?

Comment (Zeegomo) [reply waku/src/node/discovery.rs:15]: Where can I find the specs for the API? can't find it in https://rfc.vac.dev/spec/36/

Comment (danielSanchezQ) [reply waku/src/node/discovery.rs:15]: A String is used internally. As well for the url, I know a url is a `Url`, but I don't know what to use for a nameserver 😅

Comment (danielSanchezQ) [reply waku/src/node/discovery.rs]: Got a revelation in today's sleep: https://docs.rs/url/latest/url/enum.Host.html, I think that fits well.

**Pull request merged**